### DISCUSSION TITLE
Get request from script

### DISCUSSION THREAD

**Discussion by XaverStiensmeier** [TriliumNext/Trilium#4630] [Q&A] [answered]
I've tried to perform a get request within a script, but fetch never returned. Am I doing something wrong or is it somehow prevented?

```js
var url = 'https://ilaris-online.de/api/ilaris/kreatur/';

async function fetchData() {
  try {
    const response = await fetch(url);

    if (!response.ok) {
      throw new Error(`HTTP error! Status: ${response.status}`);
    }

    const kreaturen = await response.json();
    api.log("Test>");
    api.log(kreaturen);
  } catch (error) {
    api.log("Error")
    api.log(error)
  }
}

api.log("<Test");
fetchData();
```

results in: TypeError: Failed to fetch. Interestingly enough `well.not.findable` returns the expected 404 and running the script above outside of Trilium works without any issues.

**Comment by WantToLearnJapanese** [comment, marked as answer]
I guess you executed it as a frontend script.
It may be blocked due to CORS policy. Retry after changing it to a backend script.

**Reply by XaverStiensmeier** [reply]
Yes, this was it. I wasn't aware of this distinction. I should add: `api.log(kreaturen);` still didn't log anything, because log apparently can only handle strings but doesn't throw an error. Grabbing a random string from the returned json showed that everything works when fetching while in backend api.

**Reply by WantToLearnJapanese** [reply]
As far as I know the fetch process should be done as following.
```javascript
 await fetch(url).then((res)=>res.json()).then((versions)=>versions[0]);
```
Neither do I understand the full details, but what fetch returns is the Promise object. We can not directly call `.json` on Promise.

**Reply by XaverStiensmeier** [reply]
If you await the fetch like I did before, you get the object or an error before the program proceeds as far as I know.

**Reply by WantToLearnJapanese** [reply]
I have no idea then.
It would be good to check backend log. You can access it by `Left top Trilium logo`->`Advanced`->`Show backend log`.
I guess there must be some error before reaching the `api.log` line.

**Reply by XaverStiensmeier** [reply]
Perhaps I have not expressed myself clearly: Everything works except the logging of the whole json object now - thanks to your suggestion of using the backend. So I can use the json object just as expected. I think it's just that log doesn't have a string representation for json objects.

EDIT: Opening the backend log was great advice nonetheless, because I found another error in my code that way.

**Reply by zadam** [reply]
`api.log` isn't as smart as `console.log`, if you pass an object, it won't log it serialized automatically.

Instead you can do `api.log(JSON.stringify(kreaturen));`
